src/Stratis.Bitcoin.Features.Consensus.Tests/Rules/CommonRules/BitcoinActivationRuleTest.cs

```diff
@@ -37,13 +37,13 @@ public void Run_BadVersionLowerThan2_HeightHigherThanBip34_ThrowsBadVersionConse
         public void Run_BadVersionLowerThan2_HeightSameAsBip34_ThrowsBadVersionConsensusError()
         {
             this.ruleContext.Time = new DateTimeOffset(new DateTime(2017, 1, 1, 0, 1, 0));
-            var header = this.network.Consensus.ConsensusFactory.CreateBlockHeader();
+            BlockHeader header = this.network.Consensus.ConsensusFactory.CreateBlockHeader();
             this.ruleContext.ValidationContext.BlockToValidate = this.network.CreateBlock();
             header.Bits = new Target(0x1f111115);
             header.BlockTime = new DateTimeOffset(new DateTime(2017, 1, 1, 0, 1, 1));
             header.Version = 1;
 
-            this.ruleContext.ValidationContext.ChainedHeaderToValidate = new ChainedHeader(header, header.GetHash(), this.consensusRules.ConsensusParams.BuriedDeployments[BuriedDeployments.BIP34] - 1);
+            this.ruleContext.ValidationContext.ChainedHeaderToValidate = new ChainedHeader(header, header.GetHash(), this.consensusRules.ConsensusParams.BuriedDeployments[BuriedDeployments.BIP34]);
```

Comment:
what was this -1 about?

Reply:
Previously in the rule engine we were calling header being validated as ChainedHeaderToExtend so people were mislead and started supplying CT instead of correct header. 
That produced bunch of bugs (hopefully fixed already) and those +- 1 issues. 

The rule that is being tested was fixed a week ago or so.